### PR TITLE
Implement [Replace] for rust-crypto

### PR DESCRIPTION
**Pull request by sashahilton00**
Option 5 from #128, usea  replace tag to switch to avx2 branch from [awmath/rust-crypto](https://github.com/awmath/rust-crypto/tree/avx2)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -43,7 +43,7 @@ num-bigint = "0.1.35"
 protobuf = "1.1"
 rand = "0.3.13"
 rpassword = "0.3.0"
-rust-crypto = { git = "https://github.com/awmath/rust-crypto.git", branch = "avx2" }
+rust-crypto = "0.2.36"
```
Comment (plietar) [inline Cargo.toml]: This needs to be done in every crate, eg `core/Cargo.toml`.

Comment (sashahilton00) [reply Cargo.toml]: slipped my mind, will do that now.

File: audio/Cargo.toml [hunk 2]
```diff
@@ -14,12 +14,15 @@ lewton = "0.8.0"
...
 num-bigint = "0.1.35"
 num-traits = "0.1.36"
-rust-crypto = { git = "https://github.com/awmath/rust-crypto.git", branch = "avx2" }
+rust-crypto = "0.2.36"
 tempfile = "2.1"
 
 tremor = { git = "https://github.com/plietar/rust-tremor", optional = true }
 vorbis = { version ="0.1.0", optional = true }
 
+[replace]
```
Comment (plietar) [inline audio/Cargo.toml]: These should only be at the root though.

Comment (sashahilton00) [reply audio/Cargo.toml]: so [replace] propagates to dependencies?

**Comment by sashahilton00** [comment]
Fixed up the history, let me know if anything else needs adjusting.

**Comment by awiouy** [comment]
Should there not be a [replace] statement in every Cargo.toml?

**Comment by sashahilton00** [comment]
Originally I had that, but Paul says it's only necessary in the root Cargo.toml, presumably it propagates

**Comment by plietar** [comment]
From https://doc.rust-lang.org/cargo/reference/manifest.html#the-workspace-section

> The [patch] and [replace] sections in Cargo.toml are only recognized in the root crate's manifest, and ignored in member crates' manifests.

**Comment by michaelherger** [comment]
As I'm using librespot as a library for my own application, would I have to add the [replace] section to my own project's Cargo.toml, though I don't use it in there directly?

**Comment by plietar** [comment]
@michaelherger I think so. You can try with and without the `[replace]` section and see which version of rust-crypto cargo is building.

If the section is correctly defined you should get:
```
Compiling rust-crypto v0.2.36 (https://github.com/awmath/rust-crypto.git?branch=avx2#394c2472)
```

Otherwise you just get
```
Compiling rust-crypto v0.2.36
```

**Comment by michaelherger** [comment]
Interesting: it's compiling the avx2 branch, without any modification on my end. I'll keep this in mind. Just in case there were unexpected issues. Thanks!

**Pull request merged**